node/src/main.rs

```diff
@@ -334,10 +319,12 @@ async fn main() {
             node_id.clone(),
             metrics_registry.clone(),
             firehose_networks_by_kind.get(&BlockchainKind::Ethereum),
+            substreams_networks_by_kind.get(&BlockchainKind::Ethereum),
```

Comment:
This will start to be interesting because this force a `substreams` right not be for a "known" network. `substreams` brings the possibility to support an infinite number of chain as long as an `substreams` endpoint exist. 

Going full free form raises interesting question though. I think it would be a good subject for Wednesday weekly call.

Reply:
For querying there are some aspects like block resolution that require the block cache to be populated. This is why we've gone this way. By being an endpoint type, we can require the chain to be ingested and thus we can rely on the ChainStore for block resolution and "close to head" status

Reply:
Otherwise we'd need a "substreams" ingestor if that makes sense